lib/galaxy/tools/test.py

```diff
@@ -183,16 +183,25 @@ def _process_raw_inputs(
                 param_value = raw_input_dict["value"]
                 param_extra = raw_input_dict["attributes"]
                 location = param_extra.get("location")
-                if param_value is None and location:
-                    # If no value is given, we try to get the file name directly from the URL
-                    param_value = os.path.basename(location)
                 if not value.type == "text":
                     param_value = _split_if_str(param_value)
                 if isinstance(value, galaxy.tools.parameters.basic.DataToolParameter):
-                    if not isinstance(param_value, list):
-                        param_value = [param_value]
-                    for v in param_value:
-                        _add_uploaded_dataset(context.for_state(), v, param_extra, value, required_files)
+                    if param_value is None and location:
```

Comment:
That should only be valid if `value` is `multiple`. Looking at the code, I don't think this work for collections either.

Reply:
ouch, I see, where can I find an example of a tool wrapper test with collections and how should the `location` work in that case? I'm afraid I don't know much about tool wrapping and I will miss all those cases, having a good set of tests from the start will be very helpful :sweat_smile:

Reply:
https://docs.galaxyproject.org/en/latest/dev/schema.html#id135 -- it's nicer, there's no comma-separated strings :)

Reply:
Thank you @mvdbeek, I think I've addressed the missing `multiple` check and the support for collections. I've also updated the PR description with the new test including a collection.

Let me know if this can be still improved!

Reply:
Thank you @mvdbeek, I think I've addressed the missing `multiple` check and the support for collections. I've also updated the PR description with the new test including a collection.

Let me know if this can be still improved!